### PR TITLE
Re-try Minisat download (up to two times) in case it fails

### PR DESCRIPTION
**Pull request by tautschnig**
Running in CI we do see transient network access errors from time to time, which
should not result in task failures (as long as they are genuinely transient).

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a My contribution is formatted in line with CODING_STANDARD.md.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by martin-cs — APPROVED** [review]
It's a fudge but ... I can see why it's useful.

File: src/Makefile [hunk 1]
```diff
@@ -89,7 +89,12 @@ TAR = tar
 
 minisat2-download:
 	@echo "Downloading Minisat 2.2.1"
-	@$(DOWNLOADER) http://ftp.debian.org/debian/pool/main/m/minisat2/minisat2_2.2.1.orig.tar.gz
+	@for i in `seq 1 3` ; do \
+	  $(DOWNLOADER) \
+	    http://ftp.debian.org/debian/pool/main/m/minisat2/minisat2_2.2.1.orig.tar.gz && \
```
Comment (smowton) [inline src/Makefile:94]: Might want to remove a partial download after each retry

File: src/Makefile [hunk 2]
```diff
@@ -89,7 +89,12 @@ TAR = tar
...
+	    exit 0 ; \
```
Comment (smowton) [inline src/Makefile:95]: Suggest print if retrying

File: src/Makefile [hunk 3]
```diff
@@ -89,7 +89,12 @@ TAR = tar
 
 minisat2-download:
 	@echo "Downloading Minisat 2.2.1"
-	@$(DOWNLOADER) http://ftp.debian.org/debian/pool/main/m/minisat2/minisat2_2.2.1.orig.tar.gz
+	@for i in `seq 1 3` ; do \
```
Comment (smowton) [inline src/Makefile]: Nitpick: prefer `$(cmd)` to backticks

**Review by smowton — APPROVED** [review]

**Review by chrisr-diffblue — COMMENTED** [review]
I can see the desire to have this, but I have the feeling it's possibly implemented in the wrong place... Would this actually be better done in the .travis.yml, etc? As it stands this also does nothing for CMake based builds, but I appreciate that they handle the minisat download in a different way which is probably more annoying to fix up.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: f03a2ca48c5085778e9088bbc6d2395dc0218f3e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87498958

**Review by peterschrammel — COMMENTED** [review]
I agree with @chrisr-diffblue that this should be done in .travis.yml. `travis_retry cmd` is probably the way to do it: https://docs.travis-ci.com/user/common-build-problems/#timeouts-installing-dependencies

**Comment by tautschnig** [comment]
My reason for not putting it in the Travis config file was to avoid duplication: the very same code would also go in the CodeBuild specs for both Linux and Windows and (at least in my use case) a number of build rules living outside the repository. All of which currently invoke `make minisat2-download`, so I figured I'd just put it right in the Makefile. Hence, unless I hear further objections, I'd plan to implement  @smowton's suggestions but otherwise leave it in the Makefile. I am, however, happy to replicate this for CMake as well.

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: b223bfe03678621704530b97eb444d4bc0eedadb).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87819761
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: cc4347d1a29586a7f188275915e46bf3f2ae7a78).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87820352

**Pull request merged**